java/ql/lib/semmle/code/java/frameworks/spring/SpringHttp.qll

```diff
@@ -140,3 +143,179 @@ private class SpringHttpFlowStep extends SummaryModelCsv {
       ]
   }
 }
+
+private predicate specifiesContentType(SpringRequestMappingMethod method) {
+  method.getProducesExpr().(ArrayInit).getSize() != 0 or
+  not method.getProducesExpr() instanceof ArrayInit
+}
+
+private class SpringXssSink extends XSS::XssSink {
```

Comment:
This extension of the abstract class `XssSink` is tricky to maintain (and possibly lacks a reverse import at the moment?), so instead I think it's better to just do `extends DataFlow::Node` here and then add `SpringXssSink` as a case in `DefaultXssSink extends XssSink` in XSS.qll.

Reply:
Hmm, I see we also have an `XSS::XssSanitizer` and an `XSS::XssSinkBarrier` in here, so perhaps just leave as-is. I really would like some language-level support for enforcing the required reverse import, though (and I noticed that the reverse import already is in place).